tests/baselines/reference/jsDeclarationsExportAssignedConstructorFunctionWithSub.errors.txt

```diff
@@ -0,0 +1,23 @@
+tests/cases/conformance/jsdoc/declarations/jsDeclarationsExportAssignedConstructorFunctionWithSub.js(4,1): error TS9005: Declaration emit for this file requires using private name 'exports'. An explicit type annotation may unblock declaration emit.
+tests/cases/conformance/jsdoc/declarations/jsDeclarationsExportAssignedConstructorFunctionWithSub.js(5,10): error TS2339: Property 't' does not exist on type '{ "\"tests/cases/conformance/jsdoc/declarations/jsDeclarationsExportAssignedConstructorFunctionWithSub\"": typeof exports; }'.
+tests/cases/conformance/jsdoc/declarations/jsDeclarationsExportAssignedConstructorFunctionWithSub.js(8,10): error TS2339: Property 'instance' does not exist on type 'typeof exports'.
+
+
+==== tests/cases/conformance/jsdoc/declarations/jsDeclarationsExportAssignedConstructorFunctionWithSub.js (3 errors) ====
+    /**
+     * @param {number} p
+     */
+    module.exports = function (p) {
+    ~~~~~~
+!!! error TS9005: Declaration emit for this file requires using private name 'exports'. An explicit type annotation may unblock declaration emit.

```

Comment:
Wut. That seems... wrong? The function is inline and anonymous, so it should be hoisted in declaration emit into a local member named `_exports`.

For this file, I'd expect declarations like
```ts
declare function _exports(p: number): void;
declare class _exports {
  constructor(p: number);
  t: number;
}
declare namespace _exports {
  export function Sub(): void;
  export class Sub {
    instance: _exports;
  }
}
export = _exports;
```
ideally. ...I guess it's fine? I think it's the reference in `Sub` that actually would trip us up, since we can't track the rename+hoist. If `Sub` didn't have a public reference to the exported class, I'd definitely expect it to be OK.

Reply:
Yep, I think that error is wrong. This PR is just fixing the crash.

Reply:
Does the much simpler original repro of
```js
module.exports.MyClass = function() {
    this.x = 1
}
module.exports.MyClass.prototype = {
    a: function() {
        this.b = 2
    }
}
```
emit correctly? That has way less going on and I'd fully expect that to work if it doesn't crash.

Reply:
Nope:

```ts
export {};
```

Reply:
I added the simple test case (slightly modified to get rid of a couple of errors from the binder).

Reply:
☹️ I imagine we're just missing some `isJSConstructor` calls in `serializeSymbol` to handle it via the `serializeAsClass` function?

Reply:
So, there's *this* comment, which is correct as far as I can tell:

```
                            // Looks like a `module.exports.Sub = class {}` - if we serialize `symbol` as a class, the result will have no members,
                            // since the classiness is actually from the target of the effective alias the symbol is. yes. A BlockScopedVariable|Class|Property
                            // _really_ acts like an Alias, and none of a BlockScopedVariable, Class, or Property. This is the travesty of JS binding today.
```

You don't get any members. I didn't dig any deeper than to verify that the comment correctly describes the current behaviour.

Are you OK with the fix as-is? I just want to fix the crash for now.